zkvm/src/contract.rs

```diff
@@ -70,12 +70,7 @@ impl Contract {
 
 impl Input {
     pub fn from_bytes(data: Vec<u8>) -> Result<Self, VMError> {
-        let mut reader = Subslice::new(&data);
-        let contract = Self::decode(&mut reader)?;
-        // Check that reader bytes have been fully consumed
-        if reader.len() != 0 {
-            return Err(VMError::FormatError);
-        }
+        let contract = SliceReader::parse(&data, |mut r| Self::decode(&mut r))?;
```

Comment:
would that work now?

```suggestion
        let contract = SliceReader::parse(&data, |r| Self::decode(r))?;
```